### PR TITLE
Add function XIRR

### PR DESCRIPTION
**Pull request by sequba**
## Summary

Implements the Excel `XIRR` function (HF-76).

- Adds `XIRR(Values, Dates[, Guess])` to `FinancialPlugin` — returns the internal rate of return for a schedule of cash flows that is not necessarily periodic
- Optional `Guess` defaults to `0.1`; solver uses Newton–Raphson with day-based discounting on a 365-day year
- Validates aligned value/date ranges (≥2 flows, mixed signs, date ordering), coerces empty value cells to `0`, and returns `#NA!` for single-cell or scalar arguments
- Registers `XIRR` in all language packs, updates `built-in-functions.md` and `CHANGELOG.md`
- Unit tests added in the private test suite (`function-xirr.spec.ts`)

## Test plan

- [x] `npm test` — lint + unit + browser
- [x] Microsoft XIRR example, XNPV consistency, range layouts, date edge cases, error paths, guess handling, non-convergence

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [x] Additional language file, or a change to an existing language file (translations)
- [x] Change to the documentation

---

> [!NOTE]
> **Low Risk**
> Additive built-in function in FinancialPlugin with localized names and documentation; no changes to auth, persistence, or core engine APIs.
> 
> **Overview**
> Adds Excel-compatible **`XIRR(Values, Dates[, Guess])`** for irregular cash-flow schedules, alongside existing **`IRR`** and **`XNPV`**.
> 
> Implementation lives in **`FinancialPlugin`**: aligned value/date ranges are sanitized (empty value cells → `0`, errors propagated, non-numeric → `#VALUE!`), with checks for length match, at least two flows, mixed signs, valid dates, and `Guess > -1`. The rate is solved with **Newton–Raphson** on day-based discounting using a **365-day year**, including overshoot clamping when the iterate would fall at or below **-1**.
> 
> **`XIRR`** is registered in all language packs, documented in **`built-in-functions.md`**, noted in **`CHANGELOG.md`**, and **`AGENTS.md`** gains a reminder to keep PR descriptions up to date.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 36c14650863e1098edf63ecb9067576413577177. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by qunabu** [comment]
Task linked: [HF-76 Implement function XIRR](https://app.clickup.com/t/9015210959/HF-76)

File: src/interpreter/plugin/FinancialPlugin.ts [hunk 1]
```diff
@@ -788,6 +798,49 @@ export class FinancialPlugin extends FunctionPlugin implements FunctionPluginTyp
...
+          return paymentDates
+        }
+
+        if (cashFlows.length !== paymentDates.length) {
+          return new CellError(ErrorType.NUM, ErrorMessage.EqualLength)
+        }
+
+        // A schedule needs at least two cash flows to define a rate of return.
+        if (cashFlows.length < 2) {
+          return new CellError(ErrorType.NA)
```
[the remark this answers is not in the document]
Comment (sequba) [reply src/interpreter/plugin/FinancialPlugin.ts:831]: @cursor give me the concrete example that I can verify inside Excel to make sure what should be returned in such a case

[the remark this answers is not in the document]
Comment (sequba) [reply src/interpreter/plugin/FinancialPlugin.ts:831]: @cursor I checked that in Excel and it returns #N/A

**Review by marcin-kordas-hoc — APPROVED** [review]
LGTM — approving. Additive function, full i18n coverage across all packs, and the paired test suite (#21) is thorough. Two non-blocking notes:

**1. `epsMax` convergence tolerance (nit, non-blocking).** `epsMax = 1e-10` is used as an *absolute* tolerance on the NPV. For cash flows in the millions, `|npv| < 1e-10` is effectively unreachable, so convergence ends up resting entirely on the rate-delta check; for very small flows it can trip a touch early. The name also reads as a maximum, but it's a minimum tolerance — and the same constant backs three different thresholds (NPV, derivative singularity, rate-delta). Might be worth a relative NPV tolerance plus a large-magnitude test case, but nothing blocking here.

**2. `list-of-differences.md` (question).** XIRR has a couple of Excel-specific behaviors — 365-day year, and "the first date is the minimum" rather than requiring the dates to be sorted. Should we add an entry to `docs/guide/list-of-differences.md`, or is this considered fully Excel-consistent with nothing to note?

Nice work.
